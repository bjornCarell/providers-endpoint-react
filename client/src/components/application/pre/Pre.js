import React from 'react';
import PropTypes from 'prop-types';
import { PreStyled } from './PreStyled';

export const Pre = ({ JSON, fontSize, display }) => (
  <PreStyled display={display} fontSize={`${fontSize}rem`}>
    {JSON && JSON}
  </PreStyled>
);

Pre.propTypes = {
  loading: PropTypes.bool,
  JSON: PropTypes.string.isRequired,
  fontSize: PropTypes.string.isRequired,
  display: PropTypes.string.isRequired
};
