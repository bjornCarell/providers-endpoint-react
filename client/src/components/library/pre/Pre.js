import React from 'react';
import PropTypes from 'prop-types';
import { PreStyled } from './PreStyled';

export const Pre = ({ model, JSON, fontSize, display }) => (
  <PreStyled display={display} fontSize={`${fontSize}rem`}>
    {model.length > 0 && JSON}
  </PreStyled>
);

Pre.propTypes = {
  model: PropTypes.array.isRequired,
  JSON: PropTypes.string.isRequired,
  fontSize: PropTypes.string.isRequired,
  display: PropTypes.string.isRequired
};
