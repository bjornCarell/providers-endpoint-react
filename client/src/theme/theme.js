export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
