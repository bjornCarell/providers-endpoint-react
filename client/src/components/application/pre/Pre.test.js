import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Pre } from './Pre';
import { theme } from '../../../theme/theme';

const data = {
  accessType: 'OPEN_BANKING',
  authenticationUserType: 'PERSONAL',
  authenticationFlow: 'REDIRECT',
  capabilities: ['CHECKING_ACCOUNTS'],
  credentialsType: 'THIRD_PARTY_APP',
  currency: 'DKK',
  displayName: 'BankNordik',
  fields: [],
  financialInstitutionId: 'bb1f3af4840d4920bab0fe9a39e0e263',
  financialInstitutionName: 'BankNordik',
  images: {
    icon: 'https://cdn.tink.se/provider-images/fo/fo-banknordik-password.png',
    banner: null
  },
  market: 'FO',
  multiFactor: true,
  name: 'fo-banknordik-ob',
  popular: false,
  status: 'ENABLED',
  transactional: true,
  type: 'BANK'
};

const stringJSON = JSON.stringify(data);

it('renders Pre without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(
    <ThemeProvider theme={theme}>
      <Pre JSON={stringJSON} fontSize="1.8" display="block">
        {JSON}
      </Pre>
    </ThemeProvider>,
    div
  );
});

it('renders Pre with set styles', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Pre JSON={stringJSON} fontSize="1.6" display={'block'}>
          {JSON}
        </Pre>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '1.6rem');
  expect(tree).toHaveStyleRule('display', 'block');
});
