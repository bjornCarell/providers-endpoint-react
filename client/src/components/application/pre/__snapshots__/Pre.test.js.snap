// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pre with set styles 1`] = `
.c0 {
  -webkit-animation: kKWKMx 0.5s ease-in;
  animation: kKWKMx 0.5s ease-in;
  display: block;
  font-size: 1.6rem;
  margin-top: 8rem;
}

<pre
  className="c0"
  display="block"
  fontSize="1.6rem"
>
  {"accessType":"OPEN_BANKING","authenticationUserType":"PERSONAL","authenticationFlow":"REDIRECT","capabilities":["CHECKING_ACCOUNTS"],"credentialsType":"THIRD_PARTY_APP","currency":"DKK","displayName":"BankNordik","fields":[],"financialInstitutionId":"bb1f3af4840d4920bab0fe9a39e0e263","financialInstitutionName":"BankNordik","images":{"icon":"https://cdn.tink.se/provider-images/fo/fo-banknordik-password.png","banner":null},"market":"FO","multiFactor":true,"name":"fo-banknordik-ob","popular":false,"status":"ENABLED","transactional":true,"type":"BANK"}
</pre>
`;
